### ISSUE TITLE
Translated column filter results in TypeError: keys must be a string

### ISSUE THREAD

**Issue by ozon** [flask-admin/flask-admin#245]
I try to translate the column filter. I proceed as follows:

``` python
column_filters = ( Client.connections, )
...
column_labels = dict(
    connections=lazy_gettext(u'Connections'),
)
...

```

Flask give a Traceback with `TypeError: keys must be a string`

I tried to reproduce the problem with an flask-admin example.
https://gist.github.com/ozon/d05bf06fbdf08521b7df#file-gistfile1-py-L109
Traceback addet as comment.

It result in the same error. Without filter translations works the translated app fine. What do i wrong?

**Comment by manuelbua** [comment]
I encountered the same problem solved here, but can't find the fix applied here in the latest version: how is this supposed to work now? I implemented a lazystring-aware custom JSON encoder, but i would like to know the current solution on this, anyone?

**Comment by xqliu** [comment]
This issue is also happens on latest Flask-Admin (1.2.0), and seems the patch applied earlier is not working any more.

Because of this issue, it's not able to translate field labels in UI like below

```
    column_labels = {
        'supplier': lazy_gettext('supplier'),
        # 'name': lazy_gettext('supplier'),
        # 'code': lazy_gettext('supplier')
    }
```

Is there any update on this?

Thanks.

**Comment by xqliu** [comment]
The issue might be related to fix commit via 14dec4f5396c094fce6c726d7d1b7e0afee62a83

**Comment by xqliu** [comment]
Tried a fix 

Change

```
results[key] = items
```
on line 872 to 

```
results[as_unicode(key)] = items
```
Resolve the issue temporary.

I think a fix for both this issue and #840 is needed since this quick fix will revert fix to #840.
